app/Http/Requests/Client/BulkClientRequest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Http\Requests\Client;
+
+use App\Utils\Traits\BulkOptions;
+use Illuminate\Foundation\Http\FormRequest;
+use App\Models\Client;
+
+class BulkClientRequest extends FormRequest
+{
+    use BulkOptions;
+
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        if (!$this->has('action')) {
+            return false;
+        }
+
+        if (!in_array($this->action, $this->getBulkOptions(), true)) {
+            return false;
+        }
+
+        return auth()->user()->can($this->action, Client::class);
```

Comment:
currently we only support view_, edit_ create_ ENTITY ie view_client edit_client create_client

For bulk inserts the user would need create_client, i'm not sure passing $this->action is the right way to do this one.